siac/parse.go

```diff
@@ -47,8 +60,37 @@ func parseSize(strSize string) (string, error) {
 	return "", errUnableToParseSize
 }
 
-// coinUnits converts a siacoin amount to base units.
-func coinUnits(amount string) (string, error) {
+// currencyUnits converts a types.Currency to a string with human-readable
+// units. The unit used will be the largest unit that results in a value
+// greater than 1. The value is rounded to 3 decimal places.
+func currencyUnits(c types.Currency) string {
+	pico := types.SiacoinPrecision.Div(types.NewCurrency64(1e12))
+	if c.Cmp(pico) < 0 {
+		return c.String() + " H"
+	}
+
+	// iterate until we find a unit greater than c
+	mag := pico
+	unit := ""
+	for _, unit = range []string{"pS", "nS", "uS", "mS", "SC", "KS", "MS", "GS", "TS"} {
+		if c.Cmp(mag.Mul(types.NewCurrency64(1e3))) < 0 {
+			break
+		} else if unit != "TS" {
+			// don't want to perform this multiply on the last iter; that
+			// would give us 1.2345 TB instead of 1234.5 TB
+			mag = mag.Mul(types.NewCurrency64(1e3))
+		}
+	}
+
+	num := new(big.Rat).SetInt(c.Big())
+	denom := new(big.Rat).SetInt(mag.Big())
+	res, _ := new(big.Rat).Mul(num, denom.Inv(denom)).Float64()
+
+	return fmt.Sprintf("%.5g %s", res, unit)
+}
```

Comment:
Might also want some sane rounding in these functions. Not sure if the rounding is happening now, but previously 19999999999999 -> 19999 SC, where really you should be displaying 20000